### ISSUE TITLE
CustomFilter and FilterUpdater

### ISSUE THREAD

**Issue by gmonet** [glotzerlab/hoomd-blue#1359]
## Description

Dear all, I was trying to implement the Joshua's response to the following question : https://groups.google.com/g/hoomd-users/c/jFPS08UuacQ
However, I got an unexpected error when I'm trying to create a FilterUpdater with a CustomFilter.

## Script

```python
import hoomd
import hoomd
class NegativeCharge(hoomd.filter.CustomFilter):
    """Grab all particles with a negative charge."""
    def __call__(self, state):
        with state.cpu_local_snapshot as snap:
            return snap.particles.tag[snap.particles.charge < 0]

    def __hash__(self):
        return hash(self.__class__.__name__)

    def __eq__(self, other):
        return isinstance(other, self.__class__)

hoomd.update.FilterUpdater(1, [hoomd.filter.All()]) # Ok

hoomd.update.FilterUpdater(1, [NegativeCharge()]) # Not ok
```

## Output

```python
File ~/anaconda3/envs/hoomd/lib/python3.10/site-packages/hoomd/update/particle_filter.py:70, in FilterUpdater.__init__(self, trigger, filters)
     68 def __init__(self, trigger, filters):
     69     super().__init__(trigger)
---> 70     self._filters = hoomd.data.syncedlist.SyncedList(
     71         hoomd.filter.ParticleFilter,
     72         iterable=filters,
     73         to_synced_list=_GroupConverter(),
     74         attach_members=False)
...
    242         raise ValueError(f"Value {value} could not be validated.")
    243 except ValueError as verr:
--> 244     raise ValueError(f"Validation failed: {verr.args[0]}") from verr

ValueError: Validation failed: Value <__main__.NegativeCharge object at 0x7fd310c70730> could not be validated.
```
## Configuration

Platform:
- GPU
- CPU

Installation method:
- Conda package

Versions:
- Python version: 3.10.5
- **HOOMD-blue** version: 3.2.0

**Comment by b-butler** [comment]
Thanks @gmonet the validation logic being used here is not using the standard type validation which leads to the error. It should be a quick fix.

**Comment by gmonet** [comment]
Thanks for your fast feedback @b-butler !

**Comment by joaander** [comment]
Fixed in #1360.
